### ISSUE TITLE
Fix false positives unspaced operator in unit-no-unknown

### ISSUE THREAD

**Issue by vilchik-elena** [stylelint/stylelint#4098]
There are false positives when no spaces around `+` operator

Rule `unit-no-unknown`

```css
$i: 1;

.parent {
    width: $i/1.3+px; // Unexpected unknown unit "+px"
    width: $i/1.3+0px; // Unexpected unknown unit "+0px"
    width: $i/1.3 + 0px; // OK
    width: $i/1.3*1px; // OK
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "unit-no-unknown": true
  }
}
```

> Which version of stylelint are you using?  10.0.1

I used Demo to reproduce

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

e.g. "The following warnings were flagged:"

```shell
4:15
error
Unexpected unknown unit "+px" (unit-no-unknown)
5:15
error
Unexpected unknown unit "+0px" (unit-no-unknown)
```
I suppose there is a problem when parsing such code without spaces.

**Comment by evilebottnawi** [comment]
I think it is impossible to fix due parser problem :disappointed:

**Comment by vilchik-elena** [comment]
@evilebottnawi which parser is responsible for that? [postcss-scss](https://github.com/postcss/postcss-scss)?
I could report there

**Comment by hudochenkov** [comment]
@vilchik-elena we use [`postcss-value-parser`](https://github.com/TrySound/postcss-value-parser) to parse values of the property for this rule. @evilebottnawi one of developers of this parser.

**Comment by evilebottnawi** [comment]
I think we should search way to ignore this on stylelint side and stylelint-scss should has rule for this stuff (maybe already has)

**Comment by jeddy3** [comment]
Let's add a `isStandardSyntaxUnit` util to ignore "units" that start with an operator.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by rightaway** [comment]
`unit-no-unknown` has a false positive with `calc(1px+2px)`. @alexander-akait Is this one possible to fix?

```
Unexpected unknown unit "px+2px"`
```

**Comment by Mouvedia** [comment]
I don't get errors for 

```css
a {
  width: calc(1px+2px);
  width: $i/1.3+px;
  width: $i/1.3+0px;
}
```

see https://stylelint.io/demo/#N4Igxg9gJgpiBcIAkBLeACAjAbgDoDsCBDdYA3AFwHcUoKALDMIgGzAApMAHADwGoATLwCUefOnQ06jdKgD0mAHQBmPrzGUpDDPKWqADOoIBfAiAA0IAGYoWMAHJEAtnEQwezrncVgAzr4twCHwbAHMEEDJxdFwQdwoYfChfWIwAbQIJCVjfCgBPOxYUfAoAWkgQlFDSgCcYSCcXJJgoUt8-FJBM9ABdc27YmoBXO06MKKyYkHb-OQguGBqiCgga0vwIUqH8Xy4iMBbU9HwRln7o7OttsAoUYPLWMHXN7d39ltL5xeXVo5OWM7dS7bFBlDZbfAAaw2VEICHQFGGMG6pnwxkCFTCADFVk5lhEAFa+YKBWBcAKICZTXIFGBFEpHWIsZYwXKxc6XGmFYplTFVWr1CCNRKwVozMZTZkJNldNEgYxAA

One of the commits listed on https://github.com/stylelint/stylelint/commits/main/lib/rules/unit-no-unknown must have fixed it.
